app/interactors/lab_record_imports/update.rb

```diff
@@ -2,11 +2,12 @@ module LabRecordImports
   class Update
     include Interactor
 
-    def call
+    def call # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
       ActiveRecord::Base.transaction do
-        lab_record_import.update!(rows: params[:rows],
-                                  patient_id_state: :pending)
-        params.fetch(:rows, []).each_with_index do |row, index|
+        lab_record_import.rows_file.purge
+        lab_record_import.rows_file.attach(params[:rows_file])
+        lab_record_import.update!(patient_id_state: :pending)
+        JSON.parse(lab_record_import.rows_file.download).each_with_index do |row, index|
```

Comment:
The blob was attached two lines before, so apparently `params[:rows_file]` has the content already. IIRC it's a `File` in a temporary directory, so here something like this should work to avoid redownloading and putting the file into memory:
```suggestion
        rows = params[:rows_file].open { |file| JSON.load(file) }
        rows.each_with_index do |row, index|
```

Reply:
`open` worked! However, I had to perform a `read` operation before accessing the content

Reply:
A weird thing though is that I didn't manage to make it work using `blocks`. Neither the following structure worked to me:

`rows = params[:rows_file].open { |file| JSON.load(file.read) }`

Nor its counterpart:
```
rows = params[:rows_file].open do |file|
   file.read do |content|
      JSON.load(content)
   end
end

```

Reply:
No... if you do a `read` then the content is all back to memory. So, maybe `params[:rows_file]` is not a file then. Can you check what type is it?

Reply:
I just tried this in play Ruby and it works: `File.open('data.json') { |f| JSON.load(f) }`

Reply:
params[:rows_file] is a `ActionDispatch::Http::UploadedFile:0x000055f049507460`,
params[:rows_file].open is a `File`

Reply:
(To complete the story: We ended up using Kernel#open)